src/Mod/Part/Gui/Command.cpp

```diff
@@ -1516,11 +1517,17 @@ void CmdPartCrossSections::activated(int iMsg)
     Q_UNUSED(iMsg);
     Gui::TaskView::TaskDialog* dlg = Gui::Control().activeDialog();
     if (!dlg) {
-        std::vector<App::DocumentObject*> obj = Gui::Selection().getObjectsOfType
-            (Part::Feature::getClassTypeId());
+        std::vector<App::DocumentObject*> objs = Gui::Selection().getObjectsOfType(App::DocumentObject::getClassTypeId());
+        std::vector <Part::TopoShape> shapes;
+        for (std::vector<App::DocumentObject*>::iterator it = objs.begin(); it != objs.end(); ++it) {
+            Part::TopoShape shp = Part::Feature::getTopoShape(*it);
+            if (!shp.isNull()){
+                shapes.push_back(shp);
+            }
+        }

```

Comment:
This piece of code is used in several place. Maybe create a static function for it.

Reply:
Okay.